dessia_common/workflow/utils.py

```diff
@@ -65,3 +69,41 @@ def nonblock_variables_to_script(nonblock_variables, prefix, imports, imports_as
         imports_as_is.extend(nbv_script.imports_as_is)
         nbvs_str += f"{prefix}variable_{nbv_index} = {nbv_script.declaration}\n"
     return nbvs_str
+
+
+def update_imports(input_type, input_name):
+    """ Update imports based on the schema of the input. """
+    schema = get_schema(annotation=input_type, attribute=SchemaAttribute(input_name))
+    return schema.get_import_names(import_names=[])
+
+
+def generate_input_string(signature: str, sequence_length: int = 1):
+    """ Generate input as string. """
+    if sequence_length > 1:
+        instances = [f"\n\t{signature}" for _ in range(sequence_length)]
+        return f"[{','.join(instances)}\n]"
+    return f"{signature}"
+
+
+def is_object_sequence(sequence):
+    """ Checks if a sequence contains serializable objects. """
+    return all(isinstance(element, SerializableObject) for element in sequence)
+
+
+def process_value(value):
+    """ Processes a value based on its content. """
+    if not value:
+        return repr(value)
+
+    sequence_length = len(value)
```

Comment:
What if value has no len ?

Reply:
if value has no len, it's the first condition (same if value is None or [ ])

```py
    if not value:
        return repr(value)
```

Reply:
This condition is not safe enough. Objects could be considered True and have no len. here is an example : 

```py
class A:
    def __bool__(self):
        return True


a = A()


if not a:
    print("False")
else:
    print("Failing")
    print(len(a))
```

Reply:
Anyway, the fact that the statement has been removed **kinda** make all this irrelevant